test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Internal/PageActionInvokerProviderTest.cs

```diff
@@ -299,6 +356,152 @@ public void OnProvidersExecuting_UpdatesEntriesWhenActionDescriptorProviderColle
             Assert.NotSame(entry1, entry2);
         }
 
+        [Theory]
+        [InlineData(typeof(InheritsMethods), 2)]
```

Comment:
This isn't really a good candidate for theory tests. Let's split it out and call out what we expect in each of these 3 scenarios e.g. `PopulateHandlerMethodDescriptors_DiscoversHandlersFromBaseType` etc.